### ISSUE TITLE
Config option to limit parallel steps

### ISSUE THREAD

**Issue by theomessin** [earthly/earthly#1308]
When running a target that builds multiple targets that don't depend on each other, Earthly will run them in parallel. In fact, this is one of the reasons we picked Earthly over other tools. We certainly want to keep this functionality. However, when running these targets on less powerful machines (*cough cough* Macbook) the parallelisation can actually cause the steps to fail (e.g. running multiple `npm ci` in parallel). 

It would be nice to have a config to limit the number of steps that run in parallel.

**Comment by andrewsykim** [comment]
Doc update: https://github.com/earthly/earthly/pull/1327
